Client/Services/ClientStateManager.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using System.Threading;
+using Client.Clients;
+using Client.Interfaces;
+using Client.Models;
+
+namespace Client.Services
+{
+    // TODO sync working status.
+    // TODO Swap BatchDownloaderClient with newest implementation after merge.
+
+    enum Status
+    {
+        Working,
+        Done,
+        ShuttingDown
+    }
+
+    public class ClientStateManager
+    {
+        private Status _status;
+        private readonly ITaskClient _taskClient;
+        private readonly IHeartbeatController _heartbeatController;
+        private Thread _heartBeatThread;
+        private Thread _taskThread;
+        private Thread _batchThread;
+        private const int HeartbeatInMinutes = 5;
+        private const int ThreadTimeout = 100 * 60 * HeartbeatInMinutes;
+        private readonly Dictionary<string, string> _config;
+        private IBatchInterface _batchInterface;
+        private bool _shutdown = false;
+
+        public ClientStateManager(IHttpService httpService)
+        {
+            _taskClient = new TaskClient(httpService);
+            _heartbeatController = new HeartbeatClient(httpService);
+            ConfigManager configManager = new ConfigManager();
+            _batchInterface = new BatchDownloaderClient(httpService);
+            _config = configManager.GetConfig();
+        }
+
+        /// <summary>
+        /// Runs the client state manager.
+        /// This is responsible for running heartbeat-,task- and batchthreads.
+        /// </summary>
+        public void Run()
+        {
+            _heartBeatThread = new Thread(HeartbeatThreadHandler);
+            _taskThread = new Thread(TaskThreadHandler);
+            _batchThread = new Thread(BatchThreadHandler);
+            _batchThread.Start();
+            _taskThread.Start();
+            while (!_shutdown)
+            {
+            }
+        }
+
+        /// <summary>
+        /// Initiates shutdown of all threads.
+        /// </summary>
+        public void Shutdown()
+        {
+            _shutdown = true;
+        }
+
+
+        /// <summary>
+        /// Handles the downloading of completed tasks.
+        /// </summary>
+        private void BatchThreadHandler()
+        {
+            if (!_batchInterface.GetResult())
+            {
+                return;
+            }
+
+            List<Batch> batches = (List<Batch>) _batchInterface.GetBatchStatus();
```

Comment:
getBatchStatus returner en liste af BatchStatus ikke batches. 
```suggestion

            List<BatchStatus > batches = (List<BatchStatus >) _batchInterface.GetBatchStatus();
```